staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go

```diff
@@ -617,18 +917,12 @@ func newClient(useNewConnection bool) *http.Client {
 	}
 }
 
-func newGenericAPIServer(t *testing.T, keepListening bool) *GenericAPIServer {
+func newGenericAPIServer(t *testing.T, fAudit *fakeAudit, keepListening bool) *GenericAPIServer {
 	config, _ := setUp(t)
 	config.ShutdownDelayDuration = 100 * time.Millisecond
 	config.ShutdownSendRetryAfter = keepListening
-	config.BuildHandlerChainFunc = func(apiHandler http.Handler, c *Config) http.Handler {
-		handler := genericfilters.WithWaitGroup(apiHandler, c.LongRunningFunc, c.HandlerChainWaitGroup)
-		if c.ShutdownSendRetryAfter {
-			handler = genericfilters.WithRetryAfter(handler, c.lifecycleSignals.AfterShutdownDelayDuration.Signaled())
-		}
-		handler = genericapifilters.WithRequestInfo(handler, c.RequestInfoResolver)
-		return handler
-	}
+	config.AuditPolicyRuleEvaluator = fAudit
+	config.AuditBackend = fAudit
```

Comment:
I'm super happy to see the previous customization for BuildHandlerChainFunc going away...

Reply:
yes, it's more or less represents a production apiserver